### ISSUE TITLE
Solutions are "invalid" but blocks accepted on testnet

### ISSUE THREAD

**Issue by sammy007** [btcz/bitcoinz#43]
I used `https://github.com/litecoinz-project/equihashverify.git` verifier, the solutions from zhasher are invalid there but blocks accepted on testnet.

```
    it('should return true for correct solution test', function(){
          const header = Buffer('04000000c74e2b08bdd2d49b3ec0739b8e52e010fc1c27549c03a5acfb4f05c087f5370076b3691647afcc3911e194b0025186cdee91a7a3b6a6245cc512e843a82b370a0000000000000000000000000000000000000000000000000000000000000000c9a41c5bf6fb4c1f0000040000000000000000019400000000000000000000000000000000000000', 'hex');
          const solution = Buffer('05403bb2784090209e6c5dd90275cde96423c0a595ccfaf0450977494eb94087c17eee0b88c84df606c316def16465a611be0b4a50c4689e498bb8e51a0002506493a0a18a7f43b357baf71565d6f5b245296c2a1fed12c1cb9fc351ac9cae9aff6229eb', 'hex');

          var isValid = equihash.verify(header, solution, 144, 5);

          assert.equal(isValid, true);
      });
```

The block on testnet with this solution is 18756.

Please clarify how you solutions are special and how to verify them.

**Comment by egyptianbman** [comment]
That equihash verify doesn't account for the new BitcoinZ personalization string. Try this one: https://github.com/1ds/equihashverify

Note you'll need to test it as `equihash.verify(header, solution, "BitcoinZ01230123", 144, 5);`

**Comment by sammy007** [comment]
So not only K,N changed, I also must use `BitcoinZ01230123` as personalization string? Why not keep things simple and keep it the same as in canonical 200,9?

**Comment by egyptianbman** [comment]
The whole goal of this is to increase ASIC resistance. Maintaining the same personalization string as other coins does not accomplish this.

**Comment by sammy007** [comment]
ASICs actually don't care about K,N and also about personalization strings as well.

**Comment by egyptianbman** [comment]
Can you elaborate?

**Comment by sammy007** [comment]
It's the same useless trick as scrypt-n. It depends on ASIC designs but imo bitmain is not stupid to "hardcode" personalization string into hardware, it's likely upgradeable via firmware update.

**Comment by egyptianbman** [comment]
This implementation inhibits the Z9 from mining. If you have a better implementation for ASIC resistance, please share.

**Comment by sammy007** [comment]
Highly likely it inhibits it right until firmware upgrade on Z9.

**Comment by sammy007** [comment]
There are at least 2 coins switching to 144,5 and will use different string, why not make a consensus to at least keep mining software compatible?

**Comment by egyptianbman** [comment]
I've already answered your question. This is open source code and you're welcome to contribute if you have a better implementation. Otherwise, I'll consider your original issue resolved. Please let us know if you find any issues, thanks.

**Comment by renuzit** [comment]
https://www.cryptolux.org/images/b/b9/Equihash.pdf

"In this paper we solve this open problem and show how to
construct an asymmetric proof-of-work (PoW) based on a computationally
hard problem, which requires a lot of memory to generate
a proof (called ”memory-hardness” feature) but is instant
to verify. Our primary proposal Equihash is a PoW based on the
generalized birthday problem and enhanced Wagner’s algorithm
for it. We introduce the new technique of algorithm binding to
prevent cost amortization and demonstrate that possible parallel
implementations are constrained by memory bandwidth. Our
scheme has tunable and steep time-space tradeoffs, which impose
large computational penalties if less memory is used."

TL;DR - If the Z9s are able to mine 144,5 then it's possible that there are 
HUGE computational penalties for doing so. We know that most like the Z9s
only have 128 MB. 144,5 requires 700MB which is a lot more expensive to 
produce and would require a breakthrough in ASIC tech to achieve.
